From cf071db2f286496ca77486318fd0edcf42f5e280 Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:55 +0800
Subject: [PATCH] sw64: fix the number of aux entries in ARCH_DLINFO
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56UNZ

--------------------------------

There's only AT_SYSINFO_EHDR auxiliary vector so far in SW64 ELF.
It's used to store vdso base of current task.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/uapi/asm/auxvec.h | 21 +++------------------
 1 file changed, 3 insertions(+), 18 deletions(-)

diff --git a/arch/sw_64/include/uapi/asm/auxvec.h b/arch/sw_64/include/uapi/asm/auxvec.h
index 0d4baa48f2f3..309a8294be7a 100644
--- a/arch/sw_64/include/uapi/asm/auxvec.h
+++ b/arch/sw_64/include/uapi/asm/auxvec.h
@@ -2,25 +2,10 @@
 #ifndef _UAPI_ASM_SW64_AUXVEC_H
 #define _UAPI_ASM_SW64_AUXVEC_H
 
-/* Reserve these numbers for any future use of a VDSO.  */
-#define AT_SYSINFO		32
+/* VDSO location.  */
 #define AT_SYSINFO_EHDR		33
 
-/*
- * More complete cache descriptions than AT_[DIU]CACHEBSIZE.  If the
- * value is -1, then the cache doesn't exist.  Otherwise:
- *
- * bit 0-3:	  Cache set-associativity; 0 means fully associative.
- * bit 4-7:	  Log2 of cacheline size.
- * bit 8-31:	  Size of the entire cache >> 8.
- * bit 32-63:  Reserved.
- */
-
-#define AT_L1I_CACHESHAPE	34
-#define AT_L1D_CACHESHAPE	35
-#define AT_L2_CACHESHAPE	36
-#define AT_L3_CACHESHAPE	37
-
-#define AT_VECTOR_SIZE_ARCH	4 /* entries in ARCH_DLINFO */
+/* entries in ARCH_DLINFO */
+#define AT_VECTOR_SIZE_ARCH	1
 
 #endif /* _UAPI_ASM_SW64_AUXVEC_H */
-- 
2.34.1

